services/src/main/java/io/grpc/services/BinaryLog.java

```diff
@@ -244,4 +265,101 @@ public BinaryLog getLog(String fullMethodName) {
       return null;
     }
   }
+
+  /**
+   * Returns a {@link Uint128} by interpreting the first 8 bytes as the high int64 and the second
+   * 8 bytes as the low int64.
+   */
+  // TODO(zpencer): verify int64 representation with other gRPC languages
+  static Uint128 callIdToProto(byte[] bytes) {
+    Preconditions.checkArgument(bytes.length == 16);
+    ByteBuffer bb = ByteBuffer.wrap(bytes);
+    long high = bb.getLong();
+    long low = bb.getLong();
+    return Uint128.newBuilder().setHigh(high).setLow(low).build();
+  }
+
+  @VisibleForTesting
+  // TODO(zpencer): the binlog design does not specify how to actually express the peer bytes
+  static Peer socketToProto(SocketAddress address) {
+    PeerType peerType = null;
+    byte[] peerAddress = null;
+
+    if (address instanceof InetSocketAddress) {
+      InetAddress inetAddress = ((InetSocketAddress) address).getAddress();
+      if (inetAddress instanceof Inet4Address) {
+        peerType = PeerType.PEER_IPV4;
+      } else if (inetAddress instanceof Inet6Address) {
+        peerType = PeerType.PEER_IPV6;
+      } else {
+        logger.log(Level.SEVERE, "unknown type of InetSocketAddress: {}", address);
+      }
+      int port = ((InetSocketAddress) address).getPort();
+      byte[] portBytes = new byte[IP_PORT_BYTES];
+      portBytes[0] = (byte) (port & IP_PORT_UPPER_MASK);
+      portBytes[1] = (byte) (port & IP_PORT_LOWER_MASK);
+      peerAddress = Bytes.concat(inetAddress.getAddress(), portBytes);
+    } else if (address.getClass().getName().equals("io.netty.channel.unix.DomainSocketAddress")) {
```

Comment:
The toString() method seems right for all sockets that aren't ISA.

Reply:
done